go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dtables
+
+import (
+	"io"
+	"sort"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/diff"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/index"
+)
+
+// UnscopedDiffTable is a sql.Table implementation of a system table that shows which tables have
+// changed in each commit, across all branches.
+type UnscopedDiffTable struct {
```

Comment:
On a second pass, you probably want to implement sql.FilteredTable, which will let you only examine commit hashes in a where clause instead of the entire history.